src/UI/Components/Diagram/Context/EventWrapper.tsx

```diff
@@ -0,0 +1,192 @@
+
+    setDictToDisplay(JSON.parse(customEvent.detail));
+  };
+
+  /**
+   * Handles the event triggered when the user wants to edit an entity.
+   *
+   * @param {CustomEvent} event - The event object.
+   *
+   * @returns {void}
+   */
+  const handleEditEvent = (event): void => {
+    const customEvent = event as CustomEvent;
+
+    setCellToEdit(customEvent.detail);
+  };
+
+  /**
+   * Handles the event triggered when the user made update to the instance cell
+   * With removed ability to edit the related instances, we need to assert first if the instance triggered the event is in the instancesToSend map(it could be removed from the canvas)
+   *
+   * @param {CustomEvent} event - The event object.
+   *
+   * @returns {void}
+   */
+  const handleUpdateInstancesToSend = (event): void => {
+    const customEvent = event as CustomEvent;
+    const { cell, action } = customEvent.detail as {
+      cell: ServiceEntityBlock;
+      action: ActionEnum;
+    };
+
+    setInstancesToSend((prev) => {
+      //related instances aren't added to the instancesToSend map, this condition is here to prevent situation where we try to remove the related instance from the canvas and it's end up here with status to delete it from the inventory
+      if (prev.has(String(cell.id)) || action === ActionEnum.CREATE) {
+        return updateInstancesToSend(cell, action, prev);
+      }
+
+      return prev;
+    });
+  };
+
+  /**
+   * Handles updating the stencil state for the embedded entities.
+   * If the current count reach the max count, the adequate stencil element will become disabled.
+   *
+   * @param {CustomEvent} event - The event object.
+   *
+   * @returns {void}
+   */
+  const handleUpdateStencilState = (event): void => {
+    const customEvent = event as CustomEvent;
+    const eventData: { name: string; action: EmbeddedEventEnum } =
+      customEvent.detail;
+
+    //event listener doesn't get updated state outside setStencilState function, so all logic has to be done inside it
+    setStencilState((prev) => {
+      const stencilStateCopy = JSON.parse(JSON.stringify(prev));
+
+      // If the stencil doesn't exist, return the previous state - that's the case when we recurrently add related children to the canvas, these embedded entities aren't tracked
+      if (!stencilStateCopy[eventData.name]) {
+        return stencilStateCopy;
+      }
+
+      switch (eventData.action) {
+        case "add":
+          stencilStateCopy[eventData.name].current += 1;
+          break;
+        case "remove":
+          stencilStateCopy[eventData.name].current -= 1;
+          break;
+        default:
+          break;
+      }
+
+      // If the current count is more than or equal to the max count, disable the stencil of given embedded entity
+      if (
+        stencilStateCopy[eventData.name].max !== null &&
+        stencilStateCopy[eventData.name].max !== undefined &&
+        stencilStateCopy[eventData.name].current >=
```

Comment:
Can you please extract `stencilStateCopy[eventData.name]` into a variable? 
Even the `eventData.name` could idealy be stored in a variable, what does it refer to?